llm_demo/orchestrator/langchain_tools/tools.py

```diff
@@ -355,4 +387,50 @@ async def initialize_tools(client: aiohttp.ClientSession):
                         Input is always empty JSON blob. Example: {{}}
                         """,
         ),
+        StructuredTool.from_function(
+            coroutine=generate_list_seats(client),
+            name="Find Seats",
+            description="""
+                        Use this tool to find available seats for the user on a flight.
+                        User preference could include any of seat row, letter, class and type or none at all.
+                        Seat rows are are a number
+                        Seat letters can be a letter between A to F
+                        Seat type can be null for no preference, 'A' for aisle, 'W' for window and 'M' for middle.
+                        Seat class can be null for no preference, 'F' for first, 'B' for business, 'P' for premium, and 'E' for economy.
+                        The response format should be seat row and seat number. i.e. Seat 11B, 32A, ..
+                        Example:
+                        {{
+                            "airline": "AA",
+                            "flight_number": "452",
+                            "departure_airport": "LAX",
+                            "departure_time": "2024-01-01 05:50:00",
+                             "seat_row": null,
+                            "seat_letter": null,
+                            "seat_class": "P",
+                            "seat_type": null,
+                        }}
+                        Example:
+                        {{
+                            "airline": "UA",
+                            "flight_number": "1532",
+                            "departure_airport": "SFO",
+                            "departure_time": "2024-01-08 05:50:00",
+                            "seat_row": null,
+                            "seat_letter": null,
+                            "seat_class": "E",
+                            "seat_type": "W",
+                        }}
+                        Example:
+                        {{
+                            "airline": "OO",
+                            "flight_number": "6307",
+                            "departure_airport": "SFO",
+                            "departure_time": "2024-10-28 20:13:00",
+                            "seat_row": 25,
+                            "seat_letter": "B",
+                            "seat_class": null,
+                            "seat_type": null,
+                        }}
+                        """,
```

Comment:
After adding the `SeatInput` class, add `args_schema=SeatInput` here. (can refer to the other tool above. List ticket doesn't have the schema since it doesn't require any input.

Reply:
Awesome, sounds good! Done!